### PR TITLE
Describe `Trie` entities related to `Runtime`

### PR DESCRIPTION
**Pull request by Longarithm**
**Goal**

Document `Trie` entities closely related to `Runtime`.
Don't dig much into details, because there are some plans to rewrite it.

### CODE REVIEW

**Review by bowenwang1996 — COMMENTED** [review]
It seems that this PR describes some implementation details rather than how trie works on the protocol level. Therefore, I think it is better to submit this change to `/docs/architecture.md` in nearcore

File: specs/DataStructures/Trie.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+We use Merkle-Patricia Trie to store blockchain state.
+Trie is persistent, which means that insertion of new node actually leads to creation of new path to this node, and thus root of Trie after insertion will also be presented by new object.
+
+Here we describe structures closely related to Runtime.
+
+## Main structures
+
+### Trie
+
+Base structure. Almost not called directly from `Runtime`, modifications are made using `TrieUpdate`.
```
Comment (mzhangmzz) [inline specs/DataStructures/Trie.md:12]: nit: s/Almost not/Almost never

File: specs/DataStructures/Trie.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+### TrieUpdate
+
+Provides a way to access Storage and record changes to commit in the future. Update is prepared as follows:
+
+- changes are made using `set` and `remove` methods, which are added to `prospective` field,
+- call `commit` method which moves `prospective` changes to `committed`,
+- call `finalize` method which prepares `TrieChanges` and state changes based on `committed` field.
+
+Note that `finalize` does not modify `Trie` itself, it only prepares changes to be passed to `ChainStore`.
```
Comment (mzhangmzz) [inline specs/DataStructures/Trie.md:22]: I think what you meant here is `finalize` and `Trie::update` does not update the database storage. It only modifies trie nodes in memory. These changes are written to the database when `ShardTries::apply_insertions` is called

**Comment by Longarithm** [comment]
PR was moved here - https://github.com/near/nearcore/pull/4558, closing this one.

**Pull request closed without merging**